app/backend/Services/AzureComputerVisionService.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Net.Http.Headers;
+using System.Text;
+
+namespace MinimalApi.Services;
+
+public class AzureComputerVisionService(string endPoint, string apiKey)
+{
+    public record ImageEmbeddingResponse(string modelVersion, float[] vector);
+
+    // add virtual keyword to make it mockable
+    public virtual async Task<ImageEmbeddingResponse> VectorizeImageAsync(string imagePathOrUrl, CancellationToken ct = default)
+    {
+        var api = $"{endPoint}/computervision/retrieval:vectorizeImage?api-version=2023-02-01-preview&modelVersion=latest";
```

Comment:
Not a good idea to have hardcoded URLs, is there not a client available for us to consume?

Reply:
Not the best of my knowledge. The python sample also use this rest api.

Reply:
I found it with a quick search, this should be the preferred way to consume these APIs:

https://learn.microsoft.com/en-us/azure/ai-services/computer-vision/sdk/install-sdk?tabs=dotnetcli%2Cterminal%2Cwindows%2Cubuntu%2Cmaven&pivots=programming-language-csharp#install-the-image-analysis-sdk-for-c